subgraph/schema.graphql

```diff
@@ -1,53 +1,24 @@
-type UnlockEntity @entity {
-  id: ID!
-  count: BigInt!
-  lockOwner: Bytes! # address
-  newLockAddress: Bytes! # address
-}
-
-type Key @entity {
-  id: ID!
-  lock: Lock!
-  keyId: BigInt!
-  owner: KeyHolder!
-  expiration: BigInt!
-  tokenURI: String
-  createdAt: BigInt!
-}
-
-type KeyHolder @entity {
+type Lock @entity {
   id: ID!
   address: Bytes!
-  keys: [Key]! @derivedFrom(field: "owner")
-}
-
-type KeyPurchase @entity {
-  id: ID!
-  lock: Bytes!
-  purchaser: Bytes!
-  price: BigInt!
-  timestamp: BigInt!
+  name: String
+  expirationDuration: BigInt
   tokenAddress: Bytes!
+  price: BigInt!
+  lockManagers: [Bytes!]!
+  version: BigInt!
+  keys: [Key!] @derivedFrom(field: "lock")
+  createdAtBlock: BigInt
 }
 
-type LockManager @entity {
+type Key @entity {
   id: ID!
   lock: Lock!
-  address: Bytes!
-}
-
-type Lock @entity {
-  id: ID! #lock address
-  address: Bytes!
-  name: String
-  tokenAddress: Bytes
-  creationBlock: BigInt
-  price: BigInt
-  expirationDuration: BigInt
-  totalSupply: BigInt
-  maxNumberOfKeys: BigInt
-  LockManagers: [LockManager!] @derivedFrom(field: "lock")
-  keys: [Key!] @derivedFrom(field: "lock")
+  tokenId: BigInt!
   owner: Bytes!
-  version: BigInt
+  manager: Bytes
+  expiration: BigInt!
+  tokenURI: String
```

Comment:
Do you confirm that when the base is updated this will be changed for each key?

Reply:
not yet, I will add it now